crates/bevy_ecs/src/reflect/resource.rs

```diff
@@ -111,14 +113,19 @@ impl ReflectResource {
     }
 
     /// Gets the value of this [`Resource`] type from the world as a reflected reference.
-    pub fn reflect<'a>(&self, world: &'a World) -> Option<&'a dyn Reflect> {
-        (self.0.reflect)(world)
+    pub fn reflect<'w, 's>(
+        &self,
+        world: impl Into<FilteredResources<'w, 's>>,
```

Comment:
nit: I don't know if this param should be renamed. The component reflection functions use `entity` to represent the `Into<FilteredEntityRef>`, maybe we should use `resource` or `resources`?

Reply:
I was waffling on this because I think most users will pass `&World` and I wanted to document that doing so was possible... but I eventually realized I should actually *document* that doing so was possible.  So I renamed the parameter to `resources` and also added a note to the documentation that `&World` is valid.